state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -874,16 +927,24 @@ impl<T: Config<I>, I: 'static> Broadcaster<T::TargetChain> for Pallet<T, I> {
 	type ApiCall = T::ApiCall;
 	type Callback = <T as Config<I>>::BroadcastCallable;
 
-	fn threshold_sign_and_broadcast(
-		api_call: Self::ApiCall,
-	) -> (BroadcastId, ThresholdSignatureRequestId) {
+	fn threshold_sign_and_broadcast(api_call: Self::ApiCall) -> BroadcastId {
 		Self::threshold_sign_and_broadcast(api_call, None)
 	}
 
 	fn threshold_sign_and_broadcast_with_callback(
 		api_call: Self::ApiCall,
 		callback: Self::Callback,
-	) -> (BroadcastId, ThresholdSignatureRequestId) {
+	) -> BroadcastId {
 		Self::threshold_sign_and_broadcast(api_call, Some(callback))
 	}
+
+	fn threshold_sign_and_broadcast_rotation_tx(api_call: Self::ApiCall) -> BroadcastId {
+		let broadcast_id = Self::threshold_sign_and_broadcast(api_call, None);
+		let mut broadcast_barriers = <<<T as pallet::Config<I>>::TargetChain as Chain>::ChainCrypto as ChainCrypto>::maybe_broadcast_barriers_on_rotation(
+			broadcast_id,
+		);
+		broadcast_barriers.reverse();
```

Comment:
> I agree the whole reverse/last is a bit weird. What's wrong with just using `first`?

The reason to do the reverse is so that we can use ```pop()```. `there is no pop_front in Vec. I guess it is better to just use VecDeque and use ```pop_front```